monkey/infection_monkey/network/relay/utils.py

```diff
@@ -36,38 +40,35 @@ def find_server(servers: Iterable[str]) -> Optional[str]:
 
     for server in server_list:
         if server_results[server]:
-            return server
+            island_api_client = server_results[server]
+            return server, island_api_client
 
-    return None
+    return (None, None)
 
 
-def _find_island_server(servers: Iterator[str], server_status: MutableMapping[str, bool]):
+def _find_island_server(
+    servers: Iterator[str], server_status: MutableMapping[str, Optional[IIslandAPIClient]]
+):
     with suppress(StopIteration):
         server = next(servers)
         server_status[server] = _check_if_island_server(server)
 
 
-def _check_if_island_server(server: str) -> bool:
+def _check_if_island_server(server: str) -> IIslandAPIClient:
     logger.debug(f"Trying to connect to server: {server}")
 
     try:
-        requests.get(  # noqa: DUO123
-            f"https://{server}/api?action=is-up",
-            verify=False,
-            timeout=MEDIUM_REQUEST_TIMEOUT,
-        )
-
-        return True
-    except requests.exceptions.ConnectionError as err:
+        return HTTPIslandAPIClient(server)
```

Comment:
Can we use a factory so that we don't depend on a concrete implementation of `IIslandAPIClient`?

Reply:
@cakekoa We can discuss further and fix in another PR.